crates/proto/src/native_tls/tls_client_stream.rs

```diff
@@ -57,17 +57,19 @@ impl<S: Connect> TlsClientStreamBuilder<S> {
     /// # Arguments
     ///
     /// * `name_server` - IP and Port for the remote DNS resolver
+    /// * `socket_addr` - IP and port to connect from
     /// * `dns_name` - The DNS name, Subject Public Key Info (SPKI) name, as associated to a certificate
     #[allow(clippy::type_complexity)]
     pub fn build(
         self,
         name_server: SocketAddr,
+        bind_addr: Option<SocketAddr>,
```

Comment:
In builders I think the pattern is generally to put the Optional's in the builder itself. See the identity and add_ca options. We'd need a new `Option<SocketAddr>` in the `TlsClientStreamBuilder` struct.

Reply:
Done